src/client/game.cpp

```diff
@@ -3641,7 +3661,7 @@ bool Game::nodePlacement(const ItemDefinition &selected_def,
 
 	// formspec in meta
 	if (meta && !meta->getString("formspec").empty() && !input->isRandom()
-			&& !isKeyDown(KeyType::SNEAK)) {
+			&& !(isKeyDown(KeyType::SNEAK) || isKeyDown(KeyType::SNEAK_LOCK))) {
```

Comment:
I don't think we need to check for the sneak lock key here, but rather just check if the player is currently sneaking. The sneak key may be up, the sneak lock key may be up, but sneak itself may be "locked on" at this point, and I imagine the user may want to have node placement work "as if they're sneaking" instead of "as if they're holding down the sneak key or sneak lock key"

- Not a contributor, just a curious coder :)

Reply:
Same applies at line ~3689

Reply:
So something like `!(isKeyDown(KeyType::SNEAK) || g_settings->getBool("sneak_lock")))` ?

Reply:
Yeah that's what I'm thinking. Basically the check here is "is the player sneaking". Before, it was as simple as looking at whether the sneak key was down. Now it's a bit more complicated.